File: Front-End/src/services/productQuestion.service.ts
import axiosClient from "@/configurations/axios.config";
import type {
  AdminProductQuestionListResponse,
  ProductQuestionAddRequest,
  ProductQuestionAnswerAddRequest,
  ProductQuestionAnswerUpdateStatusRequest,
  ProductQuestionFilters,
  ProductQuestionListResponse,
  ProductQuestionResponse,
  ProductQuestionUpdateStatusRequest,
} from "@/types/productQuestion.type";

export const productQuestionService = {
  // Lấy danh sách câu hỏi theo slug sản phẩm
  getProductQuestionsBySlug: async (
    slug: string,
    page: number = 1,
    size: number = 5
  ) => {
    const response = await axiosClient.get<ProductQuestionListResponse>(
      `/product-questions/${slug}?page=${page}&size=${size}`
    );
    return response.data;
  },

  // Tạo câu hỏi mới
  createProductQuestion: async (data: ProductQuestionAddRequest) => {
    const response = await axiosClient.post<ProductQuestionResponse>(
      `/product-questions`,
      data
    );
    return response.data;
  },

  // Tạo phản hồi cho câu hỏi
  createProductQuestionAnswer: async (data: ProductQuestionAnswerAddRequest) => {
    const response = await axiosClient.post<ProductQuestionResponse>(
      `/product-questions/answer`,
      data
    );
    return response.data;
  },
  
}
// === PHẦN MỚI - THÊM VÀO CUỐI FILE ===

// Admin methods
export const productQuestionAdminService = {
  getAllProductQuestions: async (filters?: ProductQuestionFilters) => {
    const params = new URLSearchParams();
    
    if (filters?.page) params.append("page", filters.page.toString());
    if (filters?.size) params.append("size", filters.size.toString());
    if (filters?.status !== undefined && filters?.status !== "all") {
      params.append("status", filters.status.toString());
    }
    if (filters?.search) params.append("search", filters.search);
    if (filters?.productId) params.append("productId", filters.productId.toString());
    if (filters?.sortBy) params.append("sortBy", filters.sortBy);
    if (filters?.sortOrder) params.append("sortOrder", filters.sortOrder);

    const response = await axiosClient.get<AdminProductQuestionListResponse>(
      `/admin/product-questions?${params.toString()}`
    );
    return response.data;
  },

  getProductQuestionById: async (id: number) => {
    const response = await axiosClient.get<{ data: ProductQuestionResponse }>(
      `/admin/product-questions/${id}`
    );
    return response.data;
  },

  updateProductQuestionStatus: async (
    id: number,
    data: ProductQuestionUpdateStatusRequest
  ) => {
    const response = await axiosClient.patch<{ data: ProductQuestionResponse }>(
      `/admin/product-questions/${id}/status`,
      data
    );
    return response.data;
  },

  deleteProductQuestion: async (id: number) => {
    const response = await axiosClient.delete(
      `/admin/product-questions/${id}`
    );
    return response.data;
  },

  updateProductQuestionAnswerStatus: async (
    id: number,
    data: ProductQuestionAnswerUpdateStatusRequest
  ) => {
    const response = await axiosClient.patch<{ data: ProductQuestionResponse }>(
      `/admin/product-questions/answer/${id}/status`,
      data
    );
    return response.data;
  },

  deleteProductQuestionAnswer: async (id: number) => {
    const response = await axiosClient.delete(
      `/admin/product-questions/answer/${id}`
    );
    return response.data;
  },
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/ProductQuestionService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.productQuestion.ProductQuestionAddRequest;
import iuh.fit.ecommerce.dtos.request.productQuestion.ProductQuestionAnswerAddRequest;
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import iuh.fit.ecommerce.dtos.response.category.CategoryResponse;
import iuh.fit.ecommerce.dtos.response.productQuestion.ProductQuestionResponse;
import iuh.fit.ecommerce.dtos.response.productQuestion.ProductQuestionWithProductResponse;
import jakarta.transaction.Transactional;
import jakarta.validation.Valid;

import java.util.List;

public interface ProductQuestionService {
    ProductQuestionResponse createProductQuestion(@Valid ProductQuestionAddRequest request);

    ResponseWithPagination<List<ProductQuestionResponse>> getProductQuestionsByProductSlug(String slug, int page, int size);

    ProductQuestionResponse createProductQuestionAnswer(@Valid ProductQuestionAnswerAddRequest request);


    ResponseWithPagination<List<ProductQuestionWithProductResponse>> getAllProductQuestionsForAdmin(
            int page, int size, Boolean status, String search, Long productId, String sortBy, String sortOrder);

    @Transactional
    ProductQuestionWithProductResponse updateProductQuestionStatus(Long id, Boolean status);

    @Transactional
    void deleteProductQuestion(Long id);

    @Transactional
    ProductQuestionWithProductResponse updateProductQuestionAnswerStatus(Long answerId, Boolean status);

    @Transactional
    void deleteProductQuestionAnswer(Long answerId);
}
